core/src/main/java/org/elasticsearch/common/rounding/TimeZoneRounding.java

```diff
@@ -218,7 +219,19 @@ public byte id() {
         public long roundKey(long utcMillis) {
             long timeLocal = timeZone.convertUTCToLocal(utcMillis);
             long rounded = Rounding.Interval.roundValue(Rounding.Interval.roundKey(timeLocal, interval), interval);
-            return timeZone.convertLocalToUTC(rounded, false, utcMillis);
+            try {
+                return timeZone.convertLocalToUTC(rounded, true, utcMillis);
+            } catch (IllegalInstantException e) {
```

Comment:
@jimferenczi thanks, I looked into that method. The problem is that it takes a LocalDateTime object which we would have to construct each time from the local time in millis, and I saw that the way it determines whether the date is in a gap is also by try-catching an IllegalInstantException. So performance-wise unfortunately this doesn't buy us much, but we would need to construct another object on top of that. So I'd like to leave it as-is.